geoips/interfaces/module_based/output_checkers.py

```diff
@@ -0,0 +1,557 @@
+            Full path to product filename in the current output directory
+        ext : str, default=None
+            Extension to use as an alternative to the original file extension
+        flag : str, default=None
+            Additional identifying string to include in output diff filename
+
+        Returns
+        -------
+        out_diff_fname : str
+            Full path to output diff file.
+        """
+        from os import makedirs, getenv
+        from os.path import join, dirname, basename, exists, splitext
+
+        if not flag:
+            flag = ""
+        diffdir = join(
+            dirname(compare_product), "diff_test_output_dir_{0}".format(getenv("USER"))
+        )
+        out_diff_fname = join(
+            diffdir, "diff_test_output_" + flag + basename(output_product)
+        )
+        if not exists(diffdir):
+            makedirs(diffdir)
+        # Output jifs as tif for easy viewing.
+        if ext is not None:
+            out_diff_fname = splitext(out_diff_fname)[0] + ext
+        elif splitext(out_diff_fname)[-1] == ".jif":
+            out_diff_fname = splitext(out_diff_fname)[0] + ".png"
+        return out_diff_fname
+
+    def compare_outputs(self, compare_path, output_products):
+        """Compare the "correct" imagery found the list of current output_products.
+
+        Compares files produced in the current processing run with the list of
+        "correct" files contained in "compare_path".
+
+        Parameters
+        ----------
+        compare_path : str
+            Path to directory of "correct" products - filenames must match
+            output_products
+        output_products : list of str
+            List of strings of current output products,
+            to compare with products in compare_path
+
+        Returns
+        -------
+        int
+            Binary code: 0 if all comparisons were completed successfully.
+        """
+        global rezip
+
+        badcomps = []
+        goodcomps = []
+        missingcomps = []
+        missingproducts = []
+        compare_strings = []
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info("*** RUNNING COMPARISONS OF KNOWN OUTPUTS IN %s ***", compare_path)
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info("")
+        from glob import glob
+        from os.path import basename, exists, isdir, isfile, join
+
+        compare_basenames = [basename(yy) for yy in glob(compare_path + "/*")]
+        final_output_products = []
```

Comment:
Probably right here, if we wanted to implement gzip/gunzip for the test products (the ones stored in the repo), you'd have to loop through them all and gunzip any that end in .gz.  Then you'd probably have to have a dictionary of "rezips" to determine which of the files need to be rezipped at the end.  ie rezip[filename] = True or rezip[filename] = False